### ISSUE TITLE
`build package` fails when trying to build RPM packages 

### ISSUE THREAD

**Issue by whowutwut** [IBM/CAST#5]
Helping to validate the build process on this project before moving the code....  I ran `build package` to attempt to create the RPM packages, and I see the following failure: 

```
CPackRPM:Debug: You may consult rpmbuild logs in:
CPackRPM:Debug:    - /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/rpmbuildibm-export_layout.err
CPackRPM:Debug: *** + umask 022
+ cd /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/BUILD
+ mv /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/tmpBBroot
+ exit 0
+ umask 022
+ cd /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/BUILD
+ '[' /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout '!=' / ']'
+ rm -rf /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout
++ dirname /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout
+ mkdir -p /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux
+ mkdir /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout
+ '[' -e /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout ']'
+ rm -rf /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout
+ mv /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/tmpBBroot /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux/export_layout
+ /bin/true
error: Bad syntax: %defattr(,,,)
    Bad syntax: %defattr(,,,)
 ***
CPackRPM:Debug:    - /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/rpmbuildibm-export_layout.out
CPackRPM:Debug: *** Building target platforms: ppc64le
Building for target ppc64le
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.lQphSk
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.gNF0Pt
Processing files: ibm-export_layout-1.0.0-9474.ppc64le


RPM build errors:
 ***
...
CPack Error: Problem copying the package: /root/bluecoral/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.0.0-Linux-export_layout.rpm to /root/bluecoral/work/rpms/ibm-1.0.0-Linux-export_layout.rpm
CPack Error: Error when generating package: ibm
make: *** [package] Error 1
*** CONFIGURE FAILED  (cmd= make package partial_install   rc=512)
```

Attaching full logs. This is trying to build dev branch of bluecoral,  pulled about 30 min ago.
 
[build.package.log](https://github.com/IBM/CAST/files/1952868/build.package.log)

**Comment by whowutwut** [comment]
Using the following commands, hit exactly the same error: 

```
   67  cd bluecoral
   69  scripts/configure.pl --parallel --rpmbuild
   73  ./scripts/rebuild package noinstall
```

Since this was done on bluecoral project...  I can try it on CAST...  I don't see how it would make a difference but I just pulled down the latest changes.  

Trying to follow these steps: https://github.com/IBM/CAST/blob/master/CONTRIBUTING.md#build-instructions 

After the configure step... we have this print out in the script... it's not correct, so another issue that can be fixed in a pull request... 

 
```
Writing /root/CAST/.build/depgraph.dot...
-- Build files have been written to: /root/CAST/.build

################################################
# Build instructions can be found at /root/CAST/README
#
[root@fs2vm113 CAST]#
```
 
There's no file README, there's a README.md but it does not have the build instructions anymore.. 

If someone else can take a VM or any compute node, clone the project and build it with some steps, we can enhance that guide and provide a series of steps that works.  

Will update this post with the result of the build under the CAST repo.

**Comment by whowutwut** [comment]
Watching the build, looks like the error triggers from here: 
```
CPackRPM:Debug: You may consult rpmbuild logs in:
CPackRPM:Debug:    - /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/rpmbuildibm-export_layout.err
CPackRPM:Debug: *** + umask 022
+ cd /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/BUILD
+ mv /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/tmpBBroot
+ exit 0
+ umask 022
+ cd /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/BUILD
+ '[' /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout '!=' / ']'
+ rm -rf /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout
++ dirname /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout
+ mkdir -p /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux
+ mkdir /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout
+ '[' -e /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout ']'
+ rm -rf /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout
+ mv /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/tmpBBroot /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/ibm-1.1.0-Linux/export_layout
+ /bin/true
error: Bad syntax: %defattr(,,,)
    Bad syntax: %defattr(,,,)
 ***
CPackRPM:Debug:    - /root/CAST/work/rpms/_CPack_Packages/Linux/RPM/rpmbuildibm-export_layout.out
CPackRPM:Debug: *** Building target platforms: ppc64le
Building for target ppc64le
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.QA2Brd
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.PFhqRk
Processing files: ibm-export_layout-1.1.0-53.ppc64le


RPM build errors:
 ***
```

Looks to be in export_layout, seems to point to burst buffer...

**Comment by besawn** [comment]
I did some more investigating on Victor's build system, and I think the problem might be related to the cmake version that is being installed by scripts/prereq.pl.

This error is reported when attempting to build the export_layout rpm:
```
error: Bad syntax: %defattr(,,,)
    Bad syntax: %defattr(,,,)
```
Looking at the default spec file that is used to build that rpm, I think this is the section that is causing the error above:
```
grep defattr /root/CAST/scripts/gpl_template.specrpm
%defattr(@TMP_DEFAULT_FILE_PERMISSIONS@,@TMP_DEFAULT_USER@,@TMP_DEFAULT_GROUP@,@TMP_DEFAULT_DIR_PERMISSIONS@)
```
It looks like those variables are not getting set correctly in this environment.

On the Jenkins builds, the rpms are actually built using:
`scripts/configure.pl --rpmbuild --parallel --mincmake=3.6`

This --mincmake flag was added at some point due to usage of new cmake features that were first available in cmake 3.6, but I can't recall the details of which features were required. I suspect it is related to the missing variables above.

When I try to build with the same flag on Victor's machine, I get:
```
scripts/configure.pl --rpmbuild --parallel --mincmake=3.6
package datacenter-gpu-manager is not installed
cmake at /usr/local/bin/cmake is version = 3.5.2
cmake version 3.6 or higher is not installed on this system. 
Recommend running the scripts/prereq.pl script to install into your user environment.
Request that the system administrator run scripts/prereq.pl as root to make available to all users.
```

However, that is the version of cmake that is being installed by scripts/prereq.pl, so I think the script needs updating to install a newer version of cmake.

This is the version that is currently in use on machines that are able to build successfully:
`cmake at /usr/local/bin/cmake is version = 3.7.1`

I think @tgooding has been maintaining `scripts/configure.pl` and knows the history of what drove the cmake change from 3.5.2 to 3.7.1.
